ISISComputingGroup/ibex_gui
# Ticket2054 run diagnostics 2 page

### Description of work

Adds a run diagnostics 2 page similar to what was present under SECI (now called "detector diagnostics")

### To test

https://github.com/ISISComputingGroup/IBEX/issues/2054

You need the changes in https://github.com/ISISComputingGroup/EPICS-isisdae/pull/19 for this to work. If you're not sure, try a `caget TE:NDW1298:DAE:DIAG:ENABLE:SP`. If it returns something then you should be good to go, if not you need to update your DAE.

The spectra data will only be supplied when the DAE is in a RUNNING state. If it looks like it's all enabled but you're not getting any data back, check that your parameters are sensible. Very large numbers in "Integral upper limit" may cause the DAE to return slowly. An example of a set of parameters for a simulated DAE that demonstrate the functionality in an appropriate way might be:
Integral lower limit: 1
Integral upper limit: 500
Maximum frames: 1000

### Acceptance criteria

- [x] Detector diagnostics page looks visually ok
- [x] Detector diagnostics page displays data from the DAE correctly. See the PVs added in https://github.com/ISISComputingGroup/EPICS-isisdae/pull/19/files to see what data should be displayed (note, you have to manually "enable" the diagnostics, this is done automatically by IBEX when the DAE perspective is opened but if you're looking at raw PVs you'll have to do it manually)
- [x] Controls are databound to the control PVs, so if the GUI is restarted the controls should have the correct numbers populated in them.

---

#### Code Review

    - Manual system tests for the functionality should be added if there are no automated tests
    - Manual system tests can be removed from the template if they are covered by suitable automated tests

### Functional Tests


### Final Steps
